components/dialog/RotateEncryptionKeyDialog.vue

```diff
@@ -26,25 +28,31 @@ export default {
   async fetch() {
     const c = this.cluster;
 
-    if ( c.isRke1 && this.$store.getters['isRancher'] ) {
-      let etcdBackups = await this.$store.dispatch('rancher/findAll', { type: NORMAN.ETCD_BACKUP });
+    let etcdBackups = [];
 
+    if ( c.isRke1 && this.$store.getters['isRancher'] ) {
+      etcdBackups = await this.$store.dispatch('rancher/findAll', { type: NORMAN.ETCD_BACKUP });
       etcdBackups = etcdBackups.filter(backup => backup.clusterId === c.metadata.name);
+    }
 
-      const backups = sortBy(etcdBackups, ['created']).reverse();
-      const dateFormat = escapeHtml(this.$store.getters['prefs/get'](DATE_FORMAT));
-      const timeFormat = escapeHtml( this.$store.getters['prefs/get'](TIME_FORMAT));
+    if (c.isRke2 && this.$store.getters['isRancher']) {
+      etcdBackups = await this.$store.dispatch('management/findAll', { type: SNAPSHOT });
+      etcdBackups = etcdBackups.filter(backup => backup.clusterId === c.id);
+    }
 
-      if (backups.length > 0) {
-        const name = backups[0].id.split(':');
-        const d = day(backups[0].created).format(dateFormat);
-        const t = day(backups[0].created).format(timeFormat);
+    const backups = sortBy(etcdBackups, ['created']).reverse();
+    const dateFormat = escapeHtml(this.$store.getters['prefs/get'](DATE_FORMAT));
```

Comment:
This computation is not necessary if there's no backup, we should move it within the `if`.

Reply:
I have moved it within the if-statement.